File: DekafMain/tests/sql/SqlScriptTest.java
package org.jetbrains.dekaf.sql;



import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;



/**
 * @author Leonid Bushuev from JetBrains
 */
public class SqlScriptTest {


  @Test
  public void construct_2() {
    SqlScript script1 = new SqlScript("command1"),
              script2 = new SqlScript("command2");
    SqlScript script = new SqlScript(script1, script2);
    assertThat((Integer)script.getStatements().size()).isEqualTo((Integer)2);
    assertThat(script.getStatements().get(0).getSourceText()).isEqualTo("command1");
    assertThat(script.getStatements().get(1).getSourceText()).isEqualTo("command2");
  }

  @Test
  public void toString_contains_all_commands() {
    SqlScript script = new SqlScript("command 1", "command 2");
    final String text = script.toString();
    assertThat(text).contains("command 1")
                    .contains("command 2");
  }

}

File: dekaf-main-test/tests/sql/SqlScriptTest.kt
package org.jetbrains.dekaf.sql


import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test


/**
 * @author Leonid Bushuev from JetBrains
 */
@Tag("UnitTest")
class SqlScriptTest {

    @Test
    fun construct_2() {
        val script1 = SqlScript("command1")
        val script2 = SqlScript("command2")
        val script = SqlScript(script1, script2)
        assertThat(script.statements.size).isEqualTo(2)
        assertThat(script.statements[0].sourceText).isEqualTo("command1")
        assertThat(script.statements[1].sourceText).isEqualTo("command2")
    }

    @Test
    fun toString_contains_all_commands() {
        val script = SqlScript("command 1", "command 2")
        val text = script.toString()
        assertThat(text).contains("command 1")
                .contains("command 2")
    }

}
